concordia/tasks.py

```diff
@@ -212,11 +212,39 @@ def populate_asset_years():
     """
     Pull out date info from raw Item metadata and populate it for each Asset
     """
-    assets = Asset.objects.all()
-    for asset in assets:
-        metadata = asset.item.metadata
-        # I checked, and all of our data has only one entry in the dates list
-        date_info = metadata["item"]["dates"][0]
-        for asset_date in date_info:
-            asset.year = asset_date
-            asset.save()
+
+    asset_qs = Asset.objects.prefetch_related("item")
+
+    updated_count = 0
+
+    for asset_chunk in chunked(asset_qs, 500):
+        changed_assets = []
+
+        for asset in asset_chunk:
+            metadata = asset.item.metadata
+
+            year = None
+            for date_outer in metadata["item"]["dates"]:
+                for date_inner in date_outer.keys():
+                    if not date_inner.isdigit() or len(date_inner) != 4:
```

Comment:
This will fail because some of the dates are in the format of "1936 to 1954"

Reply:
Doesn't it just fail by logging? It ran to completion on my local dev install

Reply:
Yeah, but it won't populate the field for those assets, and I think it should, since it still works for sorting.